src/vec/mod.rs

```diff
@@ -528,13 +568,16 @@ impl<T, S: Storage> VecInner<T, S> {
         // use `ptr::write` to avoid running `T`'s destructor on the uninitialized memory
         debug_assert!(!self.is_full());
 
-        *self.buffer.borrow_mut().get_unchecked_mut(self.len) = MaybeUninit::new(item);
+        *self
+            .buffer
+            .borrow_mut()
+            .get_unchecked_mut(self.len.into_usize()) = MaybeUninit::new(item);
 
-        self.len += 1;
+        self.len += LenT::ONE;
     }
 
     /// Shortens the vector, keeping the first `len` elements and dropping the rest.
-    pub fn truncate(&mut self, len: usize) {
+    pub fn truncate(&mut self, len: LenT) {
```

Comment:
Same here and many other functions, I think it'd be weird if all these API's don't use usize.

Reply:
Okay, so revert the changes made to arguments and return types as said in the PR description. Got it.

Reply:
Done.